packages/components/src/components/chip/chip.tsx

```diff
@@ -0,0 +1,293 @@
+     * opened.
+     */
+    @State() opened: boolean = false;
+
+    /**
+     * State to store the selected options of the chip component.
+     */
+    @State() selectedOptions: Array<ChipItemEvent> = [];
+
+    /**
+     * Closing the dropdown menu when clicked outside of it.
+     */
+    @Listen('mousedown', { target: 'document' })
+    closeDropdownOnOutsideClick(event: MouseEvent) {
+        const path = event.composedPath();
+        const chipWrapper: HTMLElement = this.chip.shadowRoot.querySelector('.chip__wrapper');
+        const chipDropdown: HTMLElement = this.chip.shadowRoot.querySelector('.chip__dropdown');
+        if (!path.includes(chipDropdown) && !path.includes(chipWrapper) && this.opened) {
+            this.toggleDropdownMenu();
+        }
+    }
+
+    /**
+     * Listening to ifxChipItem event to update the selected option(s).
+     */
+    @Listen('ifxChipItem') 
+    updateSelectedOptions(event: CustomEvent<ChipItemEvent>) {
+        const eventDetail: ChipItemEvent = event.detail;
+        const previousSelection: Array<ChipItemEvent> = this.selectedOptions;
+        
+        /* Handling 'single' variant */
+        if (this.variant !== 'multi') {
+            if (eventDetail.selected) {
+                /* Closing the dropdown menu when the item is selected. */
+                this.opened = false;
+                /* Unselecting the previously selected option by traversing each item. */
+                const chipItems: NodeList = this.getChipItems();
+                chipItems.forEach((chipItem: HTMLIfxChipItemElement) => {
+                    if (chipItem.selected && chipItem !== event.target) {
+                        chipItem.chipState = {
+                            ...chipItem.chipState,
+                            emitIfxChipItem: false,
+                        }
+                        chipItem.selected = false;
+                    }
+                });
+                this.selectedOptions = [ eventDetail ];
+            } else {
+                this.selectedOptions = [];
+            }
+            this.value = this.selectedOptions[0] ? this.selectedOptions[0].value : undefined;
+        } else /* Handling 'multi' variant */ {
+            if (eventDetail.selected) {
+                this.selectedOptions = [...this.selectedOptions, eventDetail];
+            } else {
+                this.selectedOptions = this.selectedOptions.filter((option) => option.key !== eventDetail.key);
+            }
+            this.value = this.selectedOptions.map((option) => { return option.value });
+        }
+
+        /* Emitting ifxChange with the selected options. */
+        if (eventDetail.emitIfxChange){
+            this.ifxChange.emit({ previousSelection: previousSelection, 
+                                  currentSelection: this.selectedOptions});
+        }
+    }
+    
+    /**
+     * Returns the list of reference to the Chip Items.
```

Comment:
this function name also pretty well explains what it does and thus, we dont need the comment